src/components/AddBareMetalHosts/OCMClusterDetailTabContent.tsx

```diff
@@ -1,25 +1,50 @@
 import React from 'react';
-import { Modal, ModalVariant } from '@patternfly/react-core';
 import { normalizeClusterVersion } from '../../config';
 import { OcmClusterType } from './types';
 import { AddHostsClusterCreateParams, Cluster, getCluster, handleApiError } from '../../api';
-import RedirectToCluster from '../ui/RedirectToCluster';
 import { getOcmClusterId } from './utils';
 import { usePullSecretFetch } from '../fetching/pullSecret';
 import { ErrorState, LoadingState } from '../ui';
 import { addHostsClusters } from '../../api/addHostsClusters';
+import { AlertsContextProvider } from '../AlertsContextProvider';
+import { POLLING_INTERVAL } from '../../config';
+import AddBareMetalHosts from './AddBareMetalHosts';
 
-const AddBareMetalHostsDialog: React.FC<{
-  isOpen: boolean;
-  closeModal: () => {};
-  cluster: OcmClusterType;
-}> = ({ isOpen, cluster, closeModal }) => {
+type ClusterDetailTabType = {
+  key: number;
+  title: string;
+  contentId: string;
+  id: string;
+  show: boolean;
+  ref: React.RefObject<unknown>;
+};
+
+export const getOCMClusterDetailTab = (
+  tabs: ClusterDetailTabType[],
+  addBareMetalTabRef: React.RefObject<unknown>,
+  displayAddBareMetalHosts: boolean,
+): ClusterDetailTabType => ({
+  key: tabs.reduce((max, tab) => (tab.key > max ? tab.key : max), tabs[0].key) + 1,
+  title: 'Bare Metal',
+  contentId: 'addBareMetalHostsContent',
+  id: 'addBareMetalHosts',
+  show: displayAddBareMetalHosts,
+  ref: addBareMetalTabRef,
+});
+
+export const OCMClusterDetailTabContent: React.FC<{
+  cluster?: OcmClusterType;
+  isVisible: boolean;
+}> = ({ cluster, isVisible }) => {
   const [error, setError] = React.useState<string>();
-  const [redirect, setRedirect] = React.useState<Cluster['id']>();
+  const [day2Cluster, setDay2Cluster] = React.useState<Cluster | null>();
   const pullSecret = usePullSecretFetch();
 
   React.useEffect(() => {
-    if (isOpen && pullSecret) {
+    if (cluster && pullSecret && day2Cluster === undefined && isVisible) {
```

Comment:
The day2 cluster is not re-fetched when user navigates back to 'Bare Metal' tab. I think it would be good to always refresh data on navigating to the tab.

The polling seems not to stop when user navigates to another tab because the tab content component is not unmounted when switching tabs. This is probably ok.

Reply:
In OCM, all tabs are always rendered and data fetched, just the `hidden` attribute is changing.

Since this approach does not scale, let's stop at least our polling when not needed.
I have added a new commit doing that.